core/domain/user_jobs_one_off.py

```diff
@@ -81,6 +81,43 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class UserAuthModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating and populating UserAuthModel for
+    all registered users.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+
+        # Assumption: We do not have any profile user in the UserSettingsModel,
+        # because we have no method of associating profile users with a full
+        # user.
+
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, item.gae_id)
+        if (user_auth_model is not None) and (user_auth_model.id != item.id):
+            yield ('GAE_ID %s already registered with user_id %s.' % (
+                item.gae_id, user_auth_model.id), 1
```

Comment:
Maybe return the user id / gae id combination as values instead of the '1'

Reply:
Done. PTAL now.